dbt/adapters/spark/impl.py

```diff
@@ -231,8 +237,23 @@ def get_columns_in_relation(self, relation: Relation) -> List[SparkColumn]:
             # return relation's schema. if columns are empty from cache,
             # use get_columns_in_relation spark macro
             # which would execute 'describe extended tablename' query
-            rows: List[agate.Row] = super().get_columns_in_relation(relation)
-            columns = self.parse_describe_extended(relation, rows)
+            try:
+                rows: List[agate.Row] = super().get_columns_in_relation(
+                    relation
+                )
+                columns = self.parse_describe_extended(relation, rows)
+            except dbt.exceptions.RuntimeException as e:
+                # spark would throw error when table doesn't exist, where other
+                # CDW would just return and empty list,
+                # we can normalize the behavior here
```

Comment:
Ah! So this would be a functional change for v1.0. We added this error handling in https://github.com/dbt-labs/dbt-spark/pull/328, which was only included in v1.1+

Additionally, [v1.0 is reaching end of life tomorrow](https://docs.getdbt.com/docs/dbt-versions/core) (!) — so I think we'd be better off skipping this backport + patch.